src/components/AmountTextInput.js

```diff
@@ -5,64 +5,82 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React from 'react';
+import React, { useState, useRef, useEffect, forwardRef, useImperativeHandle } from 'react';
 import { StyleSheet, TextInput } from 'react-native';
+import { constants } from '@hathor/wallet-lib';
 import { getAmountParsed, getIntegerAmount } from '../utils';
 import { COLORS } from '../styles/themes';
 
-class AmountTextInput extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.inputRef = React.createRef();
-
-    // Store the text value
-    this.state = {
-      text: props.value || '',
-    };
-  }
+/**
+ * Text input component specifically for handling token amounts with BigInt validation.
+ *
+ * @param {Object} props
+ * @param {string} [props.value] - Initial input value
+ * @param {Function} props.onAmountUpdate - Callback when amount changes:
+ *                                          (text, bigIntValue) => void where text is the
+ *                                          formatted string and bigIntValue is the parsed BigInt
+ * @param {boolean} [props.allowOnlyInteger=false] - If true, only allow integer values
+ *                                                   (no decimals)
+ * @param {Object} [props.style] - Additional styles for the TextInput
+ * @param {boolean} [props.autoFocus] - Whether the input should be focused on mount
+ * @param {number} [props.decimalPlaces] - Number of decimal places to use (optional)
+ * @param {React.Ref} ref - Forwarded ref, exposes the focus() method
+ * @returns {React.ReactElement} A formatted amount input component
+ */
+const AmountTextInput = forwardRef((props, ref) => {
+  const inputRef = useRef(null);
+  const [text, setText] = useState(props.value || '');
+  const decimalPlaces = props.decimalPlaces == null
+    ? constants.DECIMAL_PLACES
```

Comment:
Why don't you get it from the redux as well? The default shouldn't be the lib constant, in my opinion.

Reply:
I don't like adding redux to components, I think they should be as "pure" as possible

But I made this a required parameter, can you please check dc72286e06195051f186828cfcf78c46bd5092ac?